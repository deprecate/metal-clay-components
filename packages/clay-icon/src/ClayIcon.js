import Component from 'metal-component';
import Soy from 'metal-soy';
import {Config} from 'metal-state';

import templates from './ClayIcon.soy.js';

/**
 * Implementation of the Metal Clay Icon.
 */
class ClayIcon extends Component {}

/**
 * State definition.
 * @static
 * @type {!Object}
 */
ClayIcon.STATE = {
  /**
	 * Id to be applied to the element.
	 * @instance
	 * @memberof ClayButton
	 * @type {?string|undefined}
	 * @default undefined
	 */
  id: Config.string(),

  /**
	 * Flag to indicate if the icon should be monospaced.
	 * @instance
	 * @memberof ClayIcon
	 * @type {?bool}
	 * @default false
	 */
  monospaced: Config.bool().value(false),

  /**
	 * The path to the SVG spritemap file containing the icons.
	 * @instance
	 * @memberof ClayIcon
	 * @type {!string}
	 * @default undefined
	 */
  spritemap: Config.string().required(),

  /**
	 * The name of the Clay SVG Icon e.g. `plus`.
	 * @instance
	 * @memberof ClayIcon
	 * @type {!string}
	 * @default undefined
	 */
  symbol: Config.string().required(),
};

Soy.register(ClayIcon, templates);

export {ClayIcon};
export default ClayIcon;
