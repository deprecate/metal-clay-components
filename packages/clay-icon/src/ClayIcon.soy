{namespace ClayIcon}

/**
 * This renders an icon element based on the Clay Icon definition.
 */
{template .render}
	{@param spritemap: string}
	{@param symbol: string}
	{@param? monospaced: bool}

	{let $classes kind="text"}
		lexicon-icon
		{sp}lexicon-icon-{$symbol}
		{if $monospaced}
			{sp}icon-monospaced
		{/if}
	{/let}

	<svg aria-hidden="true" class="{$classes}">
		<use xlink:href="{$spritemap}#{$symbol}" />
	</svg>
{/template}