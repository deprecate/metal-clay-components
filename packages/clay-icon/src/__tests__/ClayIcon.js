import ClayIcon from '../ClayIcon';

const spritemap = '../node_modules/lexicon-ux/build/images/icons/icons.svg';

let clayIcon;

describe('ClayIcon', function() {
	afterEach(() => {
		if (clayIcon) {
			clayIcon.dispose();
		}
	});

	it('should generate markup for icon `add-cell`', () => {
		clayIcon = new ClayIcon({
			spritemap: spritemap,
			symbol: 'add-cell',
		});

		expect(clayIcon).toMatchSnapshot();
	});

	it('should render a monospaced `add-cell` icon', () => {
		clayIcon = new ClayIcon({
			monospaced: true,
			spritemap: spritemap,
			symbol: 'add-cell',
		});

		expect(clayIcon).toMatchSnapshot();
	});

	it('should render a not monospaced `add-cell` icon', () => {
		clayIcon = new ClayIcon({
			monospaced: false,
			spritemap: spritemap,
			symbol: 'add-cell',
		});

		expect(clayIcon).toMatchSnapshot();
	});

	it('should render a `add-cell` icon with id', () => {
		clayIcon = new ClayIcon({
			id: 'myIcon',
			spritemap: spritemap,
			symbol: 'add-cell',
		});

		expect(clayIcon).toMatchSnapshot();
	});
});
