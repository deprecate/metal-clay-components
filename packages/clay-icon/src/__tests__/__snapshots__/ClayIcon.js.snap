// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayIcon should generate markup for icon \`add-cell\` 1`] = `
<svg aria-hidden="true" class="lexicon-icon lexicon-icon-add-cell">
  <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#add-cell"></use>
</svg>
`;

exports[`ClayIcon should render a \`add-cell\` icon with id 1`] = `
<svg aria-hidden="true" class="lexicon-icon lexicon-icon-add-cell" id="myIcon">
  <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#add-cell"></use>
</svg>
`;
